lib/src/front_end/delimited_list_builder.dart

```diff
@@ -44,8 +49,15 @@ class DelimitedListBuilder {
   /// literal, etc.
   DelimitedListBuilder(this._visitor, [this._style = const ListStyle()]);
 
-  ListPiece build() =>
-      ListPiece(_leftBracket, _elements, _blanksAfter, _rightBracket, _style);
+  /// Create the final [ListPiece] out of the added brackets, delimiters,
+  /// elements, and style.
+  ListPiece build() {
+    var blockElement = -1;
+    if (_style.allowBlockElement) blockElement = _findBlockElement();
+
+    return ListPiece(_leftBracket, _elements, _blanksAfter, _rightBracket,
```

Comment:
Are `_blockCandidates` different from `_elements`, if the `_style.allowBlockElement` is `true`?

If they are, could `blockElement` (which maybe should be named `blockElementIndex`?) potentially point to the wrong element when used in `ListPiece`? 


And if they aren't different, why do we need `_blockCandidates`?